code/obj/flock/furniture.dm

```diff
@@ -223,6 +223,19 @@
 		<br><span class='bold'>System Integrity:</span> [round((src.health_attack/src.health_max)*100)]%
 		<br><span class='bold'>###=-</span></span>"}
 
+// flockdrones can always move through
+/obj/storage/closet/flock/Crossed(atom/movable/mover)
+	. = ..()
+	var/mob/living/critter/flock/drone/drone = mover
+	if(!src.open && istype(drone) && !drone.floorrunning)
```

Comment:
Not sure if matters rn, but feel like the open check should be before, otherwise will ignore open for flock critters

Reply:
it's just the animation that gets played for drones. I don't want the animation to play if the closet is open, but they should be able to pass through anyway. Flockbits don't get animated by anything else for some reason, so this is for consistency.

Reply:
Err I guess concern was that for moving while inside a closet, will Flockdrones kick open closet, or pass through? Also, could move open check into the if check then?

Reply:
If you're in the closet and move, it'll open the closet. If the closet is closed and you move onto it, it'll play the flock squeeze animation and not open it. If the closet is open, you'll just move onto it as normal.

>Also, could move open check into the if check then?
which check?